p2p/module_test.go

```diff
@@ -151,9 +159,73 @@ func Test_Create_configureBootstrapNodes(t *testing.T) {
 }
 
 func TestP2pModule_WithHostOption_Restart(t *testing.T) {
-	ctrl := gomock.NewController(t)
+	privKey := cryptoPocket.GetPrivKeySeed(1)
+
+	peer := &typesP2P.NetworkPeer{
+		PublicKey:  privKey.PublicKey(),
+		Address:    privKey.Address(),
+		ServiceURL: testLocalServiceURL,
+	}
+
+	libp2pMockNet := mocknet.New()
+	host := newMockNetHost(t, libp2pMockNet, privKey, peer)
+
+	mod := newP2PModule(t, privKey, WithHostOption(host))
+
+	// start the module; should not create a new host
+	err := mod.Start()
+	require.NoError(t, err)
+
+	// assert initial host matches the one provided via `WithHost`
+	require.Equal(t, host, mod.host, "initial hosts don't match")
 
+	// stop the module; destroys module's host
+	err = mod.Stop()
+	require.NoError(t, err)
+
+	// assert host does *not* match after restart
+	err = mod.Start()
+	require.NoError(t, err)
+	require.NotEqual(t, host, mod.host, "post-restart hosts don't match")
+}
+
+func TestP2pModule_InvalidNonce(t *testing.T) {
 	privKey := cryptoPocket.GetPrivKeySeed(1)
+
+	peer := &typesP2P.NetworkPeer{
+		PublicKey:  privKey.PublicKey(),
+		Address:    privKey.Address(),
+		ServiceURL: testLocalServiceURL,
+	}
+
+	libp2pMockNet := mocknet.New()
+	host := newMockNetHost(t, libp2pMockNet, privKey, peer)
+
+	mod := newP2PModule(
+		t, privKey,
+		WithHostOption(host),
+	)
+	err := mod.Start()
+	require.NoError(t, err)
+
+	poktEnvelope := &messaging.PocketEnvelope{
+		Content: &anypb.Any{},
+		// 0 should be an invalid nonce value
+		Nonce: 0,
```

Comment:
Can you please add another few lines in this test to make sure the default (when we don't set the nonce at all), it is also invalid?